examples/fishnet_api/main.py

```diff
@@ -97,13 +97,43 @@ async def index():
 
 
 @app.get("/datasets")
-async def datasets() -> List[Dataset]:
-    return await Dataset.fetch_all()
+async def datasets(view_as: Optional[str], by: Optional[str]) -> str:
+    """
+    Get all datasets.
+
+    :param view_as: address of the user to view the datasets as and give additional permission information
+    :param by: address of the dataset owner to filter by
+    """
+    # TODO:
+    # - for the Owner (by) parameter:
+    #     - fetch all datasets for the owner
+    ds_by_owner = await Dataset.where_eq(owner=by)
+    ts_by_owner = ds_by_owner[0].timeseriesIDs
+    # - for the Requestor (view_as) parameter:
+    #     - fetch all timeseries for each dataset,
+
+    timeseries_records = await Timeseries.fetch(ts_by_owner)
+    timeseries_ids = timeseries_records[0].id_hash
+    #   - get all permissions for each timeseries & given requestor
+    permission_status = []
+    permission_records = await Permission.where_eq(timeseriesID=timeseries_ids, requestor=view_as)
+    for rec in permission_records:
+        permission_status.append(rec.status)
+    #   - respond with permission for dataset as approved, if all permissions are approved
+    if all(rec == PermissionStatus.GRANTED for rec in permission_status):
+        return "Dataset Available"
+    #     - respond with pending if at least one is still pending
+    elif PermissionStatus.REQUESTED in permission_status:
+        return "Dataset Requested"
+    #     - respond with denied if at lest one is denied
+    elif PermissionStatus.DENIED in permission_status:
+        return "Dataset Denied"
```

Comment:
This whole block should be executed for each fetched Dataset. Also, return the Dataset with it, not only the PermissionStatus.

If no Permissions are found, we need to signal that no Permission has been requested yet. Let's create a new Enum `DatasetPermissionStatus` with:
- NOT_REQUESTED
- REQUESTED
- GRANTED
- DENIED
 and return a `DatasetPermissionStatus` type instead of some arbitrary string.